shell/platform/android/io/flutter/plugin/common/MethodChannel.java

```diff
@@ -128,6 +132,10 @@ public void setMethodCallHandler(final MethodCallHandler handler) {
      * to be invoked by Flutter act as clients of this interface for sending results
      * back to Flutter. Invokers of Flutter methods provide implementations of this
      * interface for handling results received from Flutter.
+     *
+     * <p>All methods of this class must be called on the platform thread (Android main thread). For more details see
+     * <a href="https://github.com/flutter/engine/wiki/Threading-in-the-Flutter-Engine">Threading in the Flutter
```

Comment:
We should add a check for invocation of `nativeInvokePlatformMessageResponseCallback` on the wrong thread.  Or have the `BinaryReply` object run the reply action on the platform thread.

@chinmaygarde @matthew-carroll